### ISSUE TITLE
[a11y Design Review] Midpoint Prep Assistance: IIR: Veteran New Account Flow within Authenticated Experience, Welcome Pathway (Figma Prototypes)

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#89389]
# Initial Feedback & Recommendations

> [!NOTE]
> - **Review Date**: 7/26/2024
> - **Review Purpose**: Midpoint Review Prep
> - **Reviewed By**: CAIA A11ys 

CAIA A11ys have reviewed your Figma prototype (non-source code) files for mobile and desktop, noted below, in order to provide feedback that may help your team successfully prepare and pass your [Midpoint review ](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/midpoint-review)within the Collaboration Cycle. 

## Next Steps
See task lists, found at the bottom of this ticket, for next steps.

## Files Reviewed

- [Mobile Prototype](https://www.figma.com/proto/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?page-id=300%3A11104&node-id=300-11287&viewport=431%2C371%2C0.33&t=4reO0ZP2vtzpXG6v-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=300%3A11287&show-proto-sidebar=1)
    - [Profile Completion Flow / Mobile](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?node-id=301-13715&t=1VlTz0H2V5lqd5Y4-0)
    - [My VA + Message / Mobile](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?node-id=301-13714&t=1VlTz0H2V5lqd5Y4-0)
- [Desktop Prototype](https://www.figma.com/proto/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?page-id=300%3A11104&node-id=397-7338&viewport=710%2C-1108%2C0.29&t=fWV6H8hUlDmichZX-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=397%3A7338&show-proto-sidebar=1)
    - [Profile Completion Flow / Desktop](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?node-id=397-12106&t=1VlTz0H2V5lqd5Y4-1)
    - [My VA + Message /Desktop](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?node-id=395-7434&t=1VlTz0H2V5lqd5Y4-1)

## Annotations
The Collaboration Cycle strongly recommends you go to midpoint review with annotations in your mockup files.

### Annotations Recommendation

Please add VA Figma [annotations](https://design.va.gov/about/accessibility/accessibility-annotations) for:

- [ ] Headings
- [ ] Components
- [ ] Hidden/aria text 
- [ ] Managed focus (where does focus go after closing the [alert](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?node-id=397-11559&t=ML20EHaXvZCjnYcw-0)?)

## Capitalization

On [this frame](https://www.figma.com/design/CertVj7cu66kFv9TnseR30/Veteran-Onboarding?node-id=397-11142&t=ML20EHaXvZCjnYcw-0), don’t use all-caps for the note. 

**Impact**: All-caps is hard to read, especially for people with reading issues like dyslexia.

### Capitalization Screenshot
<img width="531" alt="Bolded note from prototype that reads: NOTE: FOR THE PURPOSES OF THIS PROTOTYPE, WE WILL GO THROUGH A SHORTER SIMULATION." src="https://github.com/user-attachments/assets/a8971e14-5ddd-49c7-a68a-16c8cc2b0b3d">

### Capitalization Recommendation
Do not use all-caps.

## Navigation

Provide clear and consistent navigation options (Consistent naming, styling, and positioning). 

### Navigation Screenshots

- **Add your contact information intro page screenshot:**
    - <img width="265" alt="Remind me later and go back link text." src="https://github.com/user-attachments/assets/6d5260d1-cc87-4b6a-9f46-c4e2d0741a0b">

- **Form flow pages screenshot:**
    - <img width="261" alt="Exit and remind me later link text." src="https://github.com/user-attachments/assets/1632b4d5-a236-4929-8743-3f88eebe460b">

### Navigation Recommendation 
Consider standardizing the content on the exit links so it’s clear to the user. Change the contact information intro page to ‘Go back and remind me later’ so the action is immediately clear.

## Reading Order / User Flow 

In the “Add contact information” flow, users are able to select a checkbox to skip adding their information. But this checkbox is below the form fields. 

**Impact**: A screen reader user would not know they can skip providing their email address, phone number, or mailing address information until they tab to the bottom of the page.   

### Reading Order / User Flow Screenshots

<img width="214" alt="Mail address screenshot. Empty checkbox next to: I live on a U.S. military base outside of the United States. Followed by link text: Learn more about military base addresses. Fields for Country, Street Address, Apartment or unit number, City, State, Postal Code are all required. The last option in this flow is a checkbox for Skip this, I don't want to provide an email address right now. Two buttons back and  save finish" src="https://github.com/user-attachments/assets/eb266951-c50f-4dbf-b916-5f3bd6843c18">

### Reading Order / User Flow Recommendation

- **Provide an immediate way for users to skip providing information:**
    - Move the “Skip” checkboxes above the entry fields, OR
    - **Use a different pattern entirely:** 
        - **Screen 1:** What contact information would you like to provide? (Checkboxes for email, phone, mailing address).
        - **Following screens:** those selected contact pieces, each on their own page, with no “skip” checkbox.

### Checkbox content on Mobile phone number screen 
The checkbox content for the phone number and mailing address screens both state “Skip this, I don’t want to provide an email address right now.”

### Content Connectivity Screenshot

<img width="249" alt="Screenshot step 2 of 3 has a field for mobile phone number with an empty field to type a phone number in. There is a checkbox under it with text that reads, skip this, I don't want to provide an email address right now." src="https://github.com/user-attachments/assets/8c8dc2e9-3ddf-4f63-acd1-58a01261fc6f">

### Content Connectivity Recommendation

The checkbox content should match the information that can be skipped. There’s an example screenshot above.

```[tasklist]
### A11Y Tasks
- [x] Create midpoint-prep design review ticket
- [x] Review prototypes as a team
- [x] Provide feedback to IIR team prior to Midpoint 7/26/2024
- [x] Maria to add feedback in comments due to GitHub access and onboarding delays 7/30/2024
- [x] Confirmed notifications (Slack and GitHub)
```

```[tasklist]
### IIR Tasks
- [ ] **Review:** CAIA Feedback provided on 7/26/22024
- [ ] **Review:** Additional CAIA Feedback provided on 7/30/2024
- [ ] **Recommendation:** implement CAIA feedback prior to Midpoint Review
- [ ] **Contact**: Contact CAIA to coordinate any support needed in preparation for Staging Review
```

**Comment by mutiaadhocteam** [comment]
**Headings**

Ensure that proper heading structure is followed. The programmatic level of the heading should match the visual importance.

Additionally, only one <h1> should exist on the page. Heading levels should not be skipped (e.g., the structure should not jump from <h1> to <h3>). Blind and visually impaired users must be aware of the intended structure of the content. While heading levels are not called out on mobile, blind and visually impaired users use headings to jump from content section to content section.

**Components**

All controls should announce announce their label alongside their purpose (e.g. “Skip this, I don’t want to provide an email address right now.” checkbox or “Save and finish” button). Blind and visually impaired users must be aware when a component is an interactive component, what the purpose of that interactive component is, and any additional changes that may occur.

Examples:

- Upon submission of the information, a success message should be automatically announced.
- Upon submission of null or invalid information, an error message should be automatically announced.
- Form fields should announce their purpose and label.
- Required fields should announce that they are required.

 **Managed focus (where does focus go after closing the alert?)**

Focus order and behavior should remain consistent from page to page to ensure the user is aware of the structure and content of the pages.. Upon activating the close button on the alert, focus should be sent to the beginning of the page.

Ensure that focus does not jump inappropriately to the end or the middle of the page upon loading.

**Comment by sara-amanda** [comment]
## Closing Ticket
- CAIA A11Y work is now completed for this ticket and handed off. 
- The remaining action items in the task list are for the IIR team.
